NSrtm.Core/FileExtensions/Pgm/PathResolvers/PgmPathResolverCaching.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using JetBrains.Annotations;
+
+namespace NSrtm.Core
+{
+    internal abstract class PgmPathResolverCaching : IPgmPathResolver
+    {
+        private readonly string _directory;
+
+        protected PgmPathResolverCaching(string directory)
+        {
+            _directory = directory;
+        }
+
+        [NotNull]
+        public string FindFilePath()
+        {
+            string filename = coordsToFilename();
```

Comment:
It seems you have copied definitely too much functionality from Hgt/other project.